### PR TITLE
uplift Go, dependencies and action runners

### PR DESCRIPTION
**Pull request by jaypipes**
Uplifts our Go version to 1.21, our pcidb and cobra dependencies to the latest version and uplifts our Github action runners appropriately.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,22 +1,22 @@
 module github.com/jaypipes/ghw
 
-go 1.19
+go 1.21
```
Comment (ffromani) [inline go.mod:3]: do we need any feature of go >= 1.20, or do one of our uplifted deps?

File: .github/workflows/fmtcheck.yml [hunk 2]
```diff
@@ -22,6 +22,7 @@ jobs:
            github.com:443
            api.github.com:443
            proxy.github.com:443
+           proxy.golang.org:443
```
Comment (ffromani) [inline .github/workflows/fmtcheck.yml:25]: why is this needed? I can't recall when the proxy was introduced but I seem to recall it was prior to 1.20

**Review by ffromani — COMMENTED** [review]
I'm about to merge but I have a couple hopefully quick questions as inline comments

File: go.mod (hunk 1 above)
Comment (jaypipes) [reply go.mod:3]: @ffromani no, not particularly. I was just keeping us up to date with more modern Go :)

File: .github/workflows/fmtcheck.yml (hunk 2 above)
Comment (ffromani) [reply .github/workflows/fmtcheck.yml:25]: I have no major objections here so I guess we can move on

File: go.mod (hunk 1 above)
Comment (ffromani) [reply go.mod:3]: I'm a fan of keeping backward compatibility as much as possible for library code, but I don't have any real objection here so let's bump it.
Perhaps next time let's try to be a tad more conservative for the sake of making life of the consumers of ghw easier

**Pull request merged**